plugins/deployment_manager/Deploy.ts

```diff
@@ -32,15 +43,11 @@ async function deployFromBuildFile<C extends Contract>(
   hre: HardhatRuntimeEnvironment,
   deployOpts: DeployOpts
 ): Promise<C> {
-  let [contractName, metadata] = getPrimaryContract(buildFile);
+  const [contractName, metadata] = getPrimaryContract(buildFile);
   const [ethersSigner] = await hre.ethers.getSigners();
   const signer = deployOpts.connect ?? ethersSigner;
-  const contractFactory = new hre.ethers.ContractFactory(metadata.abi, metadata.bin, signer);
-  debug(`Deploying ${Object.keys(buildFile.contracts)} with args`, deployArgs);
-  const contract = await contractFactory.deploy(...deployArgs);
-  const deployed = await contract.deployed();
-  debug(`Deployed ${contractName} @ ${contract.address}`);
-  return deployed as C;
+  const factory = new hre.ethers.ContractFactory(metadata.abi, metadata.bin, signer);
+  return doDeploy(contractName, factory, deployArgs, deployOpts, 'build file');
```

Comment:
are the types on the returned contract the same as before?

does this need to be `doDeploy<C>(`?

Reply:
Yeah still C extends contract and nah it figures it out from the signatures